content/blog/using-the-hpegl-hpe-greenlake-and-morpheus-terraform-providers-in-concert-to-create-provision-and-manage-hpe-pce-private-cloud-enterprise-vmaas-instances.md

```diff
@@ -0,0 +1,1494 @@
+
+* hpegl (HPE GreenLake), which interacts with the PCE Cloud Services (IAM and VMaaS) 
+* Morpheus, which interacts with the on-premise Morpheus instance or instances that are associated with the VMaaS service.  Each Morpheus instance is a separate on-premise installation of Morpheus 
+
+These two providers complement each other: 
+
+* hpegl deals with GreenLake IAM and manages VMaaS resources, including VM instances 
+* Morpheus deals underlying Morpheus resources on which the VMaaS resources – including VM instances – depend.  This covers resources such as Clouds, VM images, Node Types, Groups, Instance Types, and Instance Layouts. 
+
+The glue between these two providers is a hpegl *Data Source* called *hpegl_vmaas_morpheus_details*.  For a specific VMaaS *location* this data source will output: 
+
+* A Morpheus Access Token 
+* The URL of the Morpheus instance 
+* The time to expiry of the Access Token (in seconds) 
+
+The first two outputs can then be passed-into a Morpheus provider stanza.  See the following example where there is just one *location* for a VMaaS instance:
+
+```hcl
+provider "hpegl" {
+  vmaas {
+    location   = var.location
+    space_name = var.space
+  }
+}
+
+data "hpegl_vmaas_morpheus_details" "morpheus_details" {}
+
+provider "morpheus" {
+  url          = data.hpegl_vmaas_morpheus_details.morpheus_details.url
+  access_token = data.hpegl_vmaas_morpheus_details.morpheus_details.access_token
+}
+```
+
+For VMaaS instances with multiple locations *Provider Aliasing* can be used to tie a specific VMaaS *location* to the corresponding Morpheus instance:
+
+```hcl
+# Location 1
+provider "hpegl" {
+  vmaas {
+    location   = var.location_1
+    space_name = var.space_1
+  }
+
+  alias = "location_1"
+}
+
+data "hpegl_vmaas_morpheus_details" "location_1" {
+  provider = hpegl.location_1
+}
+
+provider "morpheus" {
+  url          = data.hpegl_vmaas_morpheus_details.location_1.url
+  access_token = data.hpegl_vmaas_morpheus_details.location_1.access_token
+
+  alias = "morpheus_location_1"
+}
+
+
+# Location 2
+provider "hpegl" {
+  vmaas {
+    location   = var.location_2
+    space_name = var.space_2
+  }
+
+  alias = "location_2"
+}
+
+data "hpegl_vmaas_morpheus_details" "location_2" {
+  provider = hpegl.location_2
+}
+
+provider "morpheus" {
+  url          = data.hpegl_vmaas_morpheus_details.location_2.url
+  access_token = data.hpegl_vmaas_morpheus_details.location_2.access_token
+
+  alias = "morpheus_location_2"
+}
+```
+
+# **Complete examples for One VMaaS Location**
+
+In the next two sections we present complete HCL (Hashicorp Configuration Language) examples for two configurations each with one VmaaS Location: 
+
+* One VMaaS Cloud, with one VMaaS instance 
+* Two VMaaS Clouds, each with one VMaaS instance for a total of two VMaaS instances 
+
+ The clouds are Vsphere clouds 
+
+In both cases: 
```

Comment:
The rendered version seemed a little off here, this line slightly indented more than above, just wondering if it would look better if you made a single of line 112 and 114?